clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2329,6 +2329,10 @@ void handleFailure(Throwable throwable) {
     @Override
     public DeleteRecordsResult deleteRecords(final Map<TopicPartition, RecordsToDelete> recordsToDelete,
                                              final DeleteRecordsOptions options) {
+        // avoid unnecessary metadata lookup
+        if(recordsToDelete.isEmpty()) {
```

Comment:
This fails checkstyle. We need a space after `if`

Reply:
Ah, thanks.